File: data/imported/knowledge/slides/cpp/merged/snippets/snippet_006.cpp
TEST(CommentsIntegrationTestNoFixture, InsertComment) {
    Presentation pres;
    auto& author = pres.comment_authors().add_author("Alice", "A");
    auto& slide = pres.slides()[0];
    auto now = std::chrono::system_clock::now();

    author.comments().add_comment("First", slide, PointF(1, 1), now);
    author.comments().add_comment("Third", slide, PointF(1, 3), now);
    author.comments().insert_comment(1, "Second", slide, PointF(1, 2), now);

    ASSERT_EQ(author.comments().size(), 3);
    EXPECT_EQ(author.comments()[1].text(), "Second");
}

File: data/imported/knowledge/slides/java/merged/snippets/snippet_006.java
@Test
    void testGetSlideComments() {
        try (Presentation pres = new Presentation()) {
            ICommentAuthor a1 = pres.getCommentAuthors().addAuthor("Alice", "A");
            ICommentAuthor a2 = pres.getCommentAuthors().addAuthor("Bob", "B");
            ISlide slide = pres.getSlides().get(0);
            LocalDateTime now = LocalDateTime.now();
            a1.getComments().addComment("Alice's", slide, new PointF(1, 1), now);
            a2.getComments().addComment("Bob's", slide, new PointF(2, 2), now);

            IComment[] allComments = slide.getSlideComments(null);
            assertThat(allComments).hasSize(2);

            IComment[] bobComments = slide.getSlideComments(a2);
            assertThat(bobComments).hasSize(1);
            assertThat(bobComments[0].getText()).isEqualTo("Bob's");
        }
    }